src/ethereum/osaka/vm/instructions/system.py

```diff
@@ -742,3 +743,52 @@ def revert(evm: Evm) -> None:
 
     # PROGRAM COUNTER
     pass
+
+
+def pay(evm: Evm) -> None:
+    """
+    Transfer ether to an account.
+
+    Parameters
+    ----------
+    evm :
+        The current EVM frame.
+    """
+    # STACK
+    to = to_address_without_mask(pop(evm.stack))
+    value = pop(evm.stack)
+
+    # GAS
+    if to in evm.accessed_addresses:
+        access_gas_cost = GAS_WARM_ACCESS
+    else:
+        evm.accessed_addresses.add(to)
```

Comment:
If the method encounters an error later, is the `to` address removed from warm addresses if it got added here?

Reply:
It will continue to be a part of the warm addresses in the current execution frame but will not be added to the warm addresses of the parent frame (contrast `incorporate_child_on_success` vs `incorporate_child_on_error` [here](https://github.com/gurukamath/execution-specs/blob/aa0bb3be926211cce791e79e312962159d935c25/src/ethereum/osaka/vm/__init__.py#L176)). This means the warmed up address play no further role in the execution.

If we are in the initial frame, the warmed addresses are not explicitly removed. However, since an error is basically the end of the execution in that case, the warmed addresses play not further role here either.